### ISSUE TITLE
Confusing about the Attribute Generation Accuracy

### ISSUE THREAD

**Issue by NNNNAI** [csmliu/STGAN#18]
Hello,csmliu!
First of all ,thank you so much for your brilliant work which is inspiring! : )

But I got some problem  when I  try  to calculate the Attribute Generation Accuracy .
In your paper,the average Attribute Generation Accuracy is about  90%. A few days ago,I  generated the sample_testing image by using the pre_trained model you provided,and measured  the Attribute Generation Accuracy by using  discriminator of the pre_trained model.

 However , the accuracy i got is far less than the accuracy you proposed in the paper.

**Comment by csmliu** [comment]
Yes, you're right, since our model can have different modification weight.
When I upload this code, I rewrite the test.py to merge three functions, i.e., `test_single_image`, `test_selected_image` and `test_multi_atts`. I got something wrong, so the default setting in this repo is not the one I used to generate the images in the paper.
Please refer to https://github.com/csmliu/STGAN/blob/master/att_classification/README.md which shows the way to obtain the accuracy in the paper.

**Comment by NNNNAI** [comment]
I measured the accuracies by using almost 20000 generated images.
The accuracies I got are listed as follow:
Bald:0.469
Bangs:0.585
Black_Hair:0.622
Blond_Hair:0.571
Brown_Hair:0.603
Bushy_Eyebrows:0.620
Eyeglasses:0.840
Male:0.484
Mouth_Slightly_Open:0.499
Mustache:0.514
No_Beard:0.664
Pale_Skin:0.557
Young:0.432

**Comment by csmliu** [comment]
Maybe. But I should remind you again that the imges generated with the default setting in this repo is not the same as I used in the paper. 
Please use the following command (given in https://github.com/csmliu/STGAN/blob/master/att_classification/README.md)
```python
python test.py --experiment_name 128 --test_int 2
python att_classification/test.py --img_dir ./output/128/sample_testing
```

**Comment by csmliu** [comment]
Have you tested on the command? Actually, it's a balance between the image quality and attribute manipulation ability.

**Comment by NNNNAI** [comment]
Sorry for replying so late~.There's just been a meeting .I got the accuracy exactly by the command :python att_classification/test.py --img_dir ./output/128/sample_testing  .

**Comment by csmliu** [comment]
Oh, I just noticed that you said `94.5%`. Maybe we're saying totally different things.
1. We train the attribute classification model, and evaluate the classification model on the `CelebA` test set. The average accuracy is `94.5%`, and this is used to evaluate the classification model, not the image generation model.
2. The images generated with `--test_int 2` should have an average classification accuracy of about `86%` (See Fig. 7 of [our paper](http://openaccess.thecvf.com/content_CVPR_2019/papers/Liu_STGAN_A_Unified_Selective_Transfer_Network_for_Arbitrary_Image_Attribute_CVPR_2019_paper.pdf)), you can get this number following the [att_classification/README.md](https://github.com/csmliu/STGAN/blob/master/att_classification/README.md)

**Comment by NNNNAI** [comment]
@csmliu 
What is the difference between the settings you used in the paper and the default setting in this repo, different hyperparameters  or a totally different training process of Generator and Discriminator?
Is the code you used in paper available?
Thanks for your attention! : ）

**Comment by csmliu** [comment]
The model and model parameters given in this repo and the paper are totally the same. The only different thing is the hyperparameter used for generating the images (and as I said before, I rewrote the test code for simplification). You can try the commands (especially the parameter `--test_ints`) given in [Testing](https://github.com/csmliu/STGAN/#testing), and some instructions are also given in [NOTE](https://github.com/csmliu/STGAN/#note) section.

**Comment by NNNNAI** [comment]
: )Oh , Thank you so much for your attention and tips,I'll try it.

**Comment by csmliu** [comment]
Hi @NNNNAI , I remember that a few days ago someone else turned me to for the accurate number of the quantitative results, here is the document I provided, you can refer to this document and check whether there is any other problem. BTW, thanks for paying attention to our work.
[quantitative results.xlsx](https://github.com/csmliu/STGAN/files/3579879/quantitative.results.xlsx)
